src/Simulation/Simulators/OpenSystemsSimulator/DataModel/ArrayConverter.cs

```diff
@@ -3,249 +3,245 @@
 
 #nullable enable
 
-using System;
-using System.Collections.Generic;
-using System.Linq;
 using System.Numerics;
 using System.Text.Json;
 using System.Text.Json.Serialization;
 using NumSharp;
 
-namespace Microsoft.Quantum.Experimental
+namespace Microsoft.Quantum.Experimental;
+
+/// <summary>
+///     Converts <see cref="NDArray" /> instances of dtype <c>double</c>
+///     and with a trailing index of length 2 to and from a JSON
+///     serialization.
+/// </summary>
+/// <remarks>
+///   <para>    
+///     The JSON representation consumed and produced by this converter is
+///     compatible with the representation used by Rust's <c>ndarray</c>
+///     and <c>serde</c> crates.
+///   </para>
+///   <para>
+///     In particular, each JSON object produced by this converter has
+///     three properties:
+///   </para>
+///   <list type="bullet">
+///     <item><c>"v"</c>: Indicates the <c>ndarray</c> format version being
+///       being used. Always <c>1</c> for this implementation.</item>
+///     <item><c>"dim"</c>: Lists the dimensions of the array being
+///       serialized. Will always contain one less dimension than the
+///       original array, as one dimension represents the real and imaginary
+///       components of each element.</item>
+///     <item><c>"data"</c>: Lists the elements of the array, with the
+///       right-most dimension varying the fastest (defined by <c>ndarray</c>
+///       as "logical order", and by NumPy as "C-ordered").</item>
+///   </list>
+/// </remarks>
+public class ComplexArrayConverter : JsonConverter<NDArray>
```

Comment:
(Up to you)
Discrepancy between the file name 
`ArrayConverter` and the class name
`ComplexArrayConverter `.

Reply:
Good point, let me rename the file. The original intent was to be more general, but wound up having to specialize to just complex arrays due to limitations in NumSharp.